### PR TITLE
Changed to be able to read user-scope datastore value via '{{ st2kv.user }}' notation.

### PR DESCRIPTION
**Pull request by userlocalhost**
## Background
The change of (https://github.com/StackStorm/st2/pull/3199) and [document of Datastore](https://docs.stackstorm.com/datastore.html#scoping-datastore-items) say StackStorm supports the I/F to get datastore value via {{ st2kv.system }} and {{ st2kv.user }}.

### Related Issue
* https://github.com/StackStorm/st2/issues/4451#issuecomment-442612125

## Problem
StackStorm didn't have an implementation to accept `{{ st2kv.user }}` expression.

## Change
This implements the processing to accept `{{ st2kv.user }}` notation and render it to user-scope value in datastore which is associated with executed user.

### CODE REVIEW

File: st2common/st2common/util/param.py [hunk 1]
```diff
@@ -81,6 +82,13 @@ def _create_graph(action_context, config):
     '''
     G = nx.DiGraph()
     system_keyvalue_context = {SYSTEM_SCOPE: KeyValueLookup(scope=FULL_SYSTEM_SCOPE)}
+
+    # If both 'user' and 'api_user' are specified, this prioritize 'api_user'
+    user = action_context['user'] if 'user' in action_context else None
+    user = action_context['api_user'] if 'api_user' in action_context else user
+    if user:
+        system_keyvalue_context[USER_SCOPE] = UserKeyValueLookup(scope=FULL_USER_SCOPE, user=user)
```
Comment (Kami) [inline st2common/st2common/util/param.py]: Hm, is this desired?

I'm fine with this approach, but we should document this behavior (something along the lines that ``st2kv.user.foo`` will only work for actions which have user context attached to and and are executed via the API or similar).

Comment (Kami) [reply st2common/st2common/util/param.py]: Besides that, I'm fine with this change.

Nice work :+1:

Comment (userlocalhost) [reply st2common/st2common/util/param.py]: Thank you very much for your review. I got it. I'll do that soon!

**Comment by userlocalhost** [comment]
Thank you again for your review and telling me advices to brush up.

> Also, what will happen if we are unable to read user from the context and action author references st2kv.user in a default value?

I've taken care of this problem by the idea that system user fills in for the one.

**Comment by userlocalhost** [comment]
I fixed the points that you mentioned, would you please review this patch, @Kami?

**Comment by userlocalhost** [comment]
Thank you very much!

File: st2common/st2common/util/param.py (hunk 1 above)
Comment (Kami) [reply st2common/st2common/util/param.py]: @userlocalhost Can you please also open st2docs PR which documents this behavior?

**Comment by Kami** [comment]
Merged, thanks.

Please also open a corresponding st2docs PR when you get a chance :)

**Comment by userlocalhost** [comment]
I think we don't have to amend st2docs because the explanation of this feature has already been described at [the Datastore page in st2docs](https://github.com/StackStorm/st2docs/blob/fe748d1400b09ca0c085cbab4095e84429676004/docs/source/datastore.rst#scoping-datastore-items).

This is an excerpt from it.
```
Also, you can refer to user variables in actions or workflows. The Jinja syntax to do so is {{st2kv.user.date_cmd}}.

Note that the notion of st2kv.user is available only when actions or workflows are run manually. The notion of st2kv.user is non-existent when actions or workflows are kicked off via rules. So the use of user scoped variables is limited to manual execution of actions or workflows.
```

I think this description is good enough to explain how to use this and what limitations are.

**Pull request merged**